core/templates/services/literally-canvas-helper.service.spec.ts

```diff
@@ -0,0 +1,285 @@
+      ],
+      id: 'e09d8a59-88d2-1714-b721-032dc017b81d'
+    };
+    var renderedSvgTag = '<polyline id="linepath-e09d8a59-88d2-1714-b721-032' +
+    'dc017b81d" fill="none" points="57,170 65.5,176.5" stroke="hsla(0, 0%, 0' +
+    '%, 1)" stroke-linecap="round" stroke-width="2"></polyline>';
+    var svgTag = null;
+    svgTag = (
+      LiterallyCanvasHelperService.linepathSVGRenderer(linepathShape));
+    expect(svgTag).toBe(renderedSvgTag);
+  });
+
+  it('should convert a polygon shapeobject to svg tag', function() {
+    var closedPolygonShape = {
+      strokeWidth: 2,
+      fillColor: 'hsla(0, 0%, 100%, 1)',
+      strokeColor: 'hsla(0, 0%, 0%, 1)',
+      isClosed: true,
+      points: [
+        {
+          x: 146,
+          y: 108,
+          color: 'hsla(0, 0%, 0%, 1)',
+          size: 2
+        },
+        {
+          x: 72,
+          y: 174,
+          color: 'hsla(0, 0%, 0%, 1)',
+          size: 1
+        },
+        {
+          x: 156,
+          y: 208,
+          color: 'hsla(0, 0%, 0%, 1)',
+          size: 2
+        },
+        {
+          x: 220,
+          y: 149,
+          color: 'hsla(0, 0%, 0%, 1)',
+          size: 2
+        }
+      ],
+      id: '89874c6a-1e67-a13d-d4e4-6fa1cabbbc58'
+    };
+    var openPolygonShape = {
+      strokeWidth: 2,
+      fillColor: 'hsla(0, 0%, 100%, 1)',
+      strokeColor: 'hsla(0, 0%, 0%, 1)',
+      isClosed: false,
+      points: [
+        {
+          x: 146,
+          y: 108,
+          color: 'hsla(0, 0%, 0%, 1)',
+          size: 1
+        },
+        {
+          x: 72,
+          y: 174,
+          color: 'hsla(0, 0%, 0%, 1)',
+          size: 2
+        },
+        {
+          x: 156,
+          y: 208,
+          color: 'hsla(0, 0%, 0%, 1)',
+          size: 2
+        },
+        {
+          x: 220,
+          y: 149,
+          color: 'hsla(0, 0%, 0%, 1)',
+          size: 2
+        }
+      ],
+      id: '89874c6a-1e67-d4e4-a13d-6fa1cabbbc58'
+    };
+    var closedSvgTag = '<polygon id="polygon-closed-89874c6a-1e67-a13d-d4e4-' +
+    '6fa1cabbbc58" fill="hsla(0, 0%, 100%, 1)" points="146,108 72.5,174.5 15' +
+    '6,208 220,149" stroke="hsla(0, 0%, 0%, 1)" stroke-width="2"></polygon>';
+    var openSvgTag = '<g id="polygon-open-89874c6a-1e67-d4e4-a13d-6fa1cabbbc' +
+    '58"><polyline fill="hsla(0, 0%, 100%, 1)" points="146.5,108.5 72,174 15' +
+    '6,208 220,149" stroke="none"></polyline><polyline fill="none" points="1' +
+    '46.5,108.5 72,174 156,208 220,149" stroke="hsla(0, 0%, 0%, 1)" stroke-w' +
+    'idth="2"></polyline></g>';
+    var svgTag1 = (
+      LiterallyCanvasHelperService.polygonSVGRenderer(closedPolygonShape));
+    expect(svgTag1).toBe(closedSvgTag);
+    var svgTag2 = (
+      LiterallyCanvasHelperService.polygonSVGRenderer(openPolygonShape));
+    expect(svgTag2).toBe(openSvgTag);
+  });
+
+  it('should fail svg validation', function() {
+    var validSvgTag = '<svg width="100" height="100"><rect id="rectangle-de5' +
```

Comment:
I think you could do more to make these three strings look more similar to each other except for the part that matters.

Reply:
Done.